stage #3/light-talks/spring-best-practices/spring-react/wisdom-pet/src/main/java/com/frankmoley/lil/wisdompet/services/ProductService.java

```diff
@@ -0,0 +1,60 @@
+package com.frankmoley.lil.wisdompet.services;
+
+import com.frankmoley.lil.wisdompet.data.entities.ProductEntity;
+import com.frankmoley.lil.wisdompet.data.repositories.ProductRepository;
+import com.frankmoley.lil.wisdompet.web.errors.NotFoundException;
+import com.frankmoley.lil.wisdompet.web.models.Product;
+import org.springframework.stereotype.Service;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class ProductService {
+
+  private final ProductRepository productRepository;
+
+  public ProductService(ProductRepository productRepository) {
+    this.productRepository = productRepository;
+  }
+
+  public List<Product> getAllProducts(){
+    Iterable<ProductEntity> productEntities = this.productRepository.findAll();
+    List<Product> products = new ArrayList<>();
+    productEntities.forEach(productEntity -> {
+      products.add(translateDbToWeb(productEntity));
+    });
+    return products;
```

Comment:
I'm totally ok with it, but I usually do
```
this.productRepository.findAll().stream()
    .map(this::translateDbToWeb)
    .toList();
```
It makes the list unmodifiable, and reduces the number of local variables